packages/smart-contracts/src/contracts/ERC20SwapToConversion.sol

```diff
@@ -1,160 +1,232 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "./lib/SafeERC20.sol";
-import "./Erc20ConversionProxy.sol";
-import "./ChainlinkConversionPath.sol";
-
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/access/Ownable.sol';
+import './lib/SafeERC20.sol';
+import './interfaces/IERC20ConversionProxy.sol';
+import './ChainlinkConversionPath.sol';
 
 interface IUniswapV2Router02 {
-  function swapTokensForExactTokens(
-    uint amountOut,
-    uint amountInMax,
-    address[] calldata path,
-    address to,
-    uint deadline
-  ) external returns (uint[] memory amounts);
+    function swapTokensForExactTokens(
+        uint256 amountOut,
+        uint256 amountInMax,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external returns (uint256[] memory amounts);
 }
 
-
 /**
  * @title ERC20SwapToConversion
  * @notice This contract swaps ERC20 tokens before paying a request thanks to a payment proxy
-  */
+ */
 contract ERC20SwapToConversion is Ownable {
-  using SafeERC20 for IERC20;
-
-  IUniswapV2Router02 public swapRouter;
-  Erc20ConversionProxy public paymentProxy;
-  ChainlinkConversionPath public chainlinkConversionPath;
-
-  constructor(address _swapRouterAddress, address _paymentProxyAddress, address _chainlinkConversionPath) {
-    swapRouter = IUniswapV2Router02(_swapRouterAddress);
-    paymentProxy = Erc20ConversionProxy(_paymentProxyAddress);
-    chainlinkConversionPath = ChainlinkConversionPath(_chainlinkConversionPath);
-  }
+    using SafeERC20 for IERC20;
+
+    IUniswapV2Router02 public swapRouter;
+    ChainlinkConversionPath public chainlinkConversionPath;
+
+    uint256 public requestSwapFees;
+    address public requestFeesCollector;
+
+    constructor(
+        address _swapRouterAddress,
+        address _chainlinkConversionPath,
+        address _owner,
+        address _requestFeesCollector,
+        uint256 _requestSwapFees
```

Comment:
It's easier to reduce the constructor interface to a minimum (ability to factorize deployment scripts and ease of post-deployment verification or debugging). So I think you should trime `_requestFeesCollector` and `requestSwapFees` at least, and maybe even the 2 first addresses.

Reply:
Done, and I removed the `requestFeesCollector` address as discussed: it will be the same as the one passed as a parameter.